src/math/iir_df2t_hifi3.c

```diff
@@ -131,4 +131,32 @@ int32_t iir_df2t(struct iir_state_df2t *iir, int32_t x)
 	return out;
 }
 
+int16_t iir_df2t_s16(struct iir_state_df2t *iir, int16_t x)
+{
+	ae_f32x2 y = iir_df2t(iir, ((int32_t)x) << 16);
+
+	return AE_ROUND16X4F32SSYM(y, y);
+}
+
+int32_t iir_df2t_s24(struct iir_state_df2t *iir, int32_t x)
+{
+	ae_f32x2 y = iir_df2t(iir, x << 8);
+
+	return AE_SRAI32R(y, 8);
+}
+
+int16_t iir_df2t_s32_s16(struct iir_state_df2t *iir, int32_t x)
+{
+	ae_f32x2 y = iir_df2t(iir, x);
+
+	return AE_ROUND16X4F32SSYM(y, y);
+}
+
+int32_t iir_df2t_s32_s24(struct iir_state_df2t *iir, int32_t x)
+{
+	ae_f32x2 y = iir_df2t(iir, x);
+
+	return AE_SRAI32R(y, 8);
```

Comment:
Here's a mistake. This instruction does not saturate to 24 bits. Need in addition to shift left with saturation by 8, shift right by 8. I think my cmocka test for 24 bit treated S24_LE as S32_LE so it can't detect overflow in 24 bits.